### PR TITLE
Import user column type

### PR DESCRIPTION
**Pull request by adrinr**
## Description
Allowing importing and exporting tables with column types

### Creating a new internal table with CSV data
image

### Importing a CSV into an existing internal table
image

### Importing a CSV into an existing external table
image

### CODE REVIEW

File: packages/server/src/utilities/schema.ts [hunk 1]
```diff
@@ -155,3 +188,43 @@ export function parse(rows: Rows, schema: Schema): Rows {
     return parsedRow
   })
 }
+
+function isValidBBReference(
```
Comment (mike12345567) [inline packages/server/src/utilities/schema.ts:192]: Can this be simplified, to just:
```js
const isValid = (userIdArray: string[]) => {
  const found = userIdArray.find(userId => !db.isGlobalUserID(d._id))
  return !found
}
switch (columnSubType) {
  case FieldSubtype.USER:
    if (typeof columnData !== "string") {
      return false
    }
    const userIdArray = parseCsvExport(columnData)
    if (userIdArray.length > 1) {
      return false
    }
    return isValid(userIdArray)
  case FieldSubtype.USERS:
    if (!Array.isArray(columnData)) {
      return false
    }
    return isValid(columnData)
}
```

Comment (adrinr) [reply packages/server/src/utilities/schema.ts:192]: Both types require parsing the `columnData`, as it comes as CSV string

**Review by mike12345567 — APPROVED** [review]
LGTM!

File: packages/server/src/utilities/schema.ts [hunk 2]
```diff
@@ -155,3 +188,32 @@ export function parse(rows: Rows, schema: Schema): Rows {
...
+  columnSubtype: FieldSubtype
+): boolean {
+  switch (columnSubtype) {
+    case FieldSubtype.USER:
+    case FieldSubtype.USERS:
+      if (typeof columnData !== "string") {
+        return false
+      }
+      const userArray = parseCsvExport<{ _id: string }[]>(columnData)
+      if (columnSubtype === FieldSubtype.USER && userArray.length > 1) {
```
[the remark this answers is not in the document]
Comment (mike12345567) [reply packages/server/src/utilities/schema.ts]: Looks good!

**Review by mike12345567 — APPROVED** [review]
Yeap - looks good - gets get it in!

**Pull request merged**